link.go

```diff
@@ -30,9 +30,6 @@ import (
 // This is because Link is often desirable to be able to use as a golang map key,
 // and in that context, pointers would not result in the desired behavior.
 type Link interface {
-	// Prototype should return a LinkPrototype which carries the information
-	// to make more Link values similar to this one (but with different hashes).
-	Prototype() LinkPrototype
```

Comment:
The main thing I'd be uncomfortable with here is now this interface says effectively nothing.

I could accidentally use anything that has a `String() string` method as a Link.

There's already a very uncomfortable practical gap in what these interfaces can actually enforce vs what has to be feature-detected and casted in the functions that compose a LinkSystem, but it's still nice imo to have an interface that has enough entropy in it to defend you from most typos.  Without this function, it wouldn't defend from many typos at all.

Reply:
i guess you either end up with a prototype() of nil when the system doesn't understand the link as it's own, or you could have a `(LinkPrototype, error)` response?